cluster-autoscaler/estimator/binpacking_estimator.go

```diff
@@ -21,38 +21,34 @@ import (
 
 	apiv1 "k8s.io/api/core/v1"
 	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
-	"k8s.io/autoscaler/cluster-autoscaler/simulator/clustersnapshot"
-	"k8s.io/autoscaler/cluster-autoscaler/simulator/predicatechecker"
+	cactx "k8s.io/autoscaler/cluster-autoscaler/context"
 	"k8s.io/autoscaler/cluster-autoscaler/utils/scheduler"
 	klog "k8s.io/klog/v2"
 	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
 )
 
 // BinpackingNodeEstimator estimates the number of needed nodes to handle the given amount of pods.
 type BinpackingNodeEstimator struct {
-	predicateChecker       predicatechecker.PredicateChecker
-	clusterSnapshot        clustersnapshot.ClusterSnapshot
+	ctx                    *cactx.AutoscalingContext
```

Comment:
Feel free to ignore this comment if it's very hard to implement, but it would be much more consistent with the rest of the codebase to have ctx as a first parameter to Estimate() rather than keep it here. 
Also it seems counterintuitive to persist context. Our implementation happens to re-use same context object rather than recreating it between loops, but I'd rather not have that as an assumption if we can avoid it.

Reply:
Moved to `Estimate()` - your are absolutely right, context should not be stored.

Reply:
And I couldn't agree more about "the love for EstimationContext". If you look closer: 
 1st: this more of a Limiter / Threshold Context 
 2nd: this is the only reason why we have this builder pattern, so the threshold limits could be updated inline

This should be simplified, but as you pointed out, that's out of PR scope.